### PR TITLE
Pypicloud upload bucket

### PR DESCRIPTION
**Pull request by geophphrie**
Mods to pypicloud to allow a different bucket prefix for uploads vs cache fallback packages.

### CODE REVIEW

File: .ackrc [hunk 1]
```diff
@@ -1,4 +1,4 @@
---ignore-ack-defaults
+# --ignore-ack-defaults
```
Comment (benthuffine) [inline .ackrc:1]: Did you intend to commit your .ackrc file?

File: pypicloud/storage/gcs.py [hunk 2]
```diff
@@ -172,8 +172,22 @@ def list(self, factory=Package):
...
+                pkg.origin = "fallback" if self.upload_prefix else None
                 yield pkg
 
+        # If we have an upload_prefix, now go back and process anything that matches.
+        if self.upload_prefix:
+            blobs = self.bucket.list_blobs(prefix=self.upload_prefix)
+            for blob in blobs:
+                pkg = self.package_from_object(blob, factory)
+                if pkg is not None:
+                    # If we have a separate upload prefix, flag THIS package as being a fallback. Otherwise
```
Comment (benthuffine) [inline pypicloud/storage/gcs.py:186]: should be "flag THIS package as being an _upload_" rather than "fallback"

**Review by benthuffine — COMMENTED** [review]
Just a couple of comments. Looks good overall.

File: pypicloud/storage/gcs.py (hunk 2 above)
Comment (geophphrie) [reply pypicloud/storage/gcs.py:186]: Actually it's correct. We use the presence of an upload_prefix as the discriminator to determine whether the differentiation feature is even active. If we are in the default mode of non-discrimination, we don't set pkg.origin at all (even though the result would simply be that all packages would be set to "fallback").

**Pull request closed without merging**